examples/multi-rule/main.tf

```diff
@@ -0,0 +1,255 @@
+  # if an existing resource group is not set (null) create a new one using prefix
+  resource_group_name          = var.resource_group == null ? "${var.prefix}-resource-group" : null
+  existing_resource_group_name = var.resource_group
+}
+
+##############################################################################
+# VPC
+##############################################################################
+
+resource "ibm_is_vpc" "example_vpc" {
+  name           = "${var.prefix}-vpc"
+  resource_group = module.resource_group.resource_group_id
+  tags           = var.resource_tags
+}
+
+resource "ibm_is_subnet" "testacc_subnet" {
+  name                     = "${var.prefix}-subnet"
+  vpc                      = ibm_is_vpc.example_vpc.id
+  zone                     = "${var.region}-1"
+  total_ipv4_address_count = 256
+  resource_group           = module.resource_group.resource_group_id
+}
+
+
+##############################################################################
+# Get Cloud Account ID
+##############################################################################
+
+data "ibm_iam_account_settings" "iam_account_settings" {
+}
+
+##############################################################################
+# Create CBR Zone
+##############################################################################
+
+module "cbr_zone" {
+  source           = "terraform-ibm-modules/cbr/ibm//modules/cbr-zone-module"
+  version          = "1.9.0"
+  name             = "${var.prefix}-VPC-network-zone"
+  zone_description = "CBR Network zone containing VPC"
+  account_id       = data.ibm_iam_account_settings.iam_account_settings.account_id
+  addresses = [{
+    type  = "vpc", # to bind a specific vpc to the zone
+    value = ibm_is_vpc.example_vpc.crn,
+  }]
+}
+
+module "cos_instance" {
+  source                        = "terraform-ibm-modules/cos/ibm"
+  version                       = "6.12.2"
+  resource_group_id             = module.resource_group.resource_group_id
+  region                        = var.region
+  create_cos_instance           = true
+  create_cos_bucket             = false
+  skip_iam_authorization_policy = true
+  cos_instance_name             = "${var.prefix}-cos-instance"
+}
+
+module "cos_buckets" {
+  source  = "terraform-ibm-modules/cos/ibm//modules/buckets"
+  version = "6.12.2"
+  bucket_configs = [
+    {
+      bucket_name            = "${var.prefix}-bucket-1"
+      kms_encryption_enabled = false
+      region_location        = var.region
+      resource_group_id      = module.resource_group.resource_group_id
+      resource_instance_id   = module.cos_instance.cos_instance_id
+    },
+    {
+      bucket_name            = "${var.prefix}-bucket-2"
+      kms_encryption_enabled = false
+      region_location        = var.region
+      resource_group_id      = module.resource_group.resource_group_id
+      resource_instance_id   = module.cos_instance.cos_instance_id
+    }
+  ]
+}
+
+locals {
+  # Filter out empty tags
```

Comment:
I'm a little lost trying to figure out whats going on with all this bucket tag logic. Can you maybe add some more code comments which explain why and what the code is doing?